File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>();
        Stack<TreeNode> stack = new Stack<>();
        stack.push(root);

        while(!stack.isEmpty()){
            TreeNode curr = stack.pop();

            if(curr.left != null) stack.push(curr.left);
            if(curr.right != null) stack.push(curr.right);

            maxHeap.offer(curr.val);
        }

        while(!maxHeap.isEmpty() && k > 1){
            maxHeap.poll();
            k--;
        }
        return maxHeap.peek();
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        if not root:
            return res
        dq = deque()

        dq.append(root)

        while dq:
            nodes = []
            for _ in range(len(dq)):

                curr = dq.popleft()

                if curr.left:
                    dq.append(curr.left)
                if curr.right:
                    dq.append(curr.right)
                nodes.append(curr.val)
            res.append(nodes)
        return res